driver-mapping/src/main/java/com/datastax/driver/mapping/MappingManager.java

```diff
@@ -218,57 +219,33 @@ public Session getSession() {
 
     @SuppressWarnings("unchecked")
     private <T> Mapper<T> getMapper(Class<T> klass) {
-        Mapper<T> mapper = (Mapper<T>) mappers.get(klass);
-        if (mapper == null) {
-            synchronized (mappers) {
-                mapper = (Mapper<T>) mappers.get(klass);
-                if (mapper == null) {
-                    EntityMapper<T> entityMapper = AnnotationParser.parseEntity(klass, ReflectionMapper.factory(), this);
-                    mapper = new Mapper<T>(this, klass, entityMapper);
-                    Map<Class<?>, Mapper<?>> newMappers = new HashMap<Class<?>, Mapper<?>>(mappers);
-                    newMappers.put(klass, mapper);
-                    mappers = newMappers;
-                }
-            }
+        if (!mappers.containsKey(klass)) {
+            EntityMapper<T> entityMapper = AnnotationParser.parseEntity(klass, ReflectionMapper.factory(), this);
+            Mapper<T> mapper = new Mapper<T>(this, klass, entityMapper);
+            mappers.putIfAbsent(klass, mapper);
         }
-        return mapper;
+        return (Mapper<T>) mappers.get(klass);
```

Comment:
I might be nitpicking but I think the following would be still more efficient:

```java
        Mapper<T> mapper = (Mapper<T>) mappers.get(klass);
        if (mapper == null) {
            EntityMapper<T> entityMapper = ...;
            mapper = new Mapper<T>(this, klass, entityMapper);
            Mapper<T> old = (Mapper<T>) mappers.putIfAbsent(klass, mapper);
            if (old != null)
                mapper = old;
        }
        return mapper;
```
... because in the nominal case (i.e. there is already an entry in the map) there would be just one access to the map instead of two.